src/dune_engine/action_exec.ml

```diff
@@ -333,6 +333,14 @@ let rec exec t ~ectx ~eenv =
           (fun () ->
             if mode = Binary then
               User_error.raise
+                ~annots:
```

Comment:
Why only do it for the binary case?

Reply:
Thanks for noticing. It is now added to non binary cases as well.